### PR TITLE
Transform Extension Collider and Bounds Calculation Performance Update

### PR DESCRIPTION
**Pull request by StephenHodgson**
## Overview

Continuation of https://github.com/XRTK/XRTK-Core/pull/448

## Changes:

- update all the usages of `Transform.SetCollidersActive`
- only update transform bounds if collider is set

### CODE REVIEW

File: XRTK.SDK/Packages/com.xrtk.sdk/Features/Input/Handlers/ManipulationHandler.cs [hunk 1]
```diff
@@ -482,23 +482,31 @@ public Collider Collider
...
                 if (thisCollider == null)
                 {
                     thisCollider = gameObject.EnsureComponent<BoxCollider>();
+                    isSet = true;
                 }
 
-                transform.GetColliderBounds();
+                if (isSet)
+                {
+                    transform.GetColliderBounds(ref cachedColliders);
```
Comment (StephenHodgson) [inline XRTK.SDK/Packages/com.xrtk.sdk/Features/Input/Handlers/ManipulationHandler.cs]: We should probably call this on enable as well

Comment (tomzorz) [reply XRTK.SDK/Packages/com.xrtk.sdk/Features/Input/Handlers/ManipulationHandler.cs]: Sure, but also - do we need to run this in the Getter? I'm feeling weird about putting potentially (relatively) long-running code there. People don't assume getters do a lot

Comment (tomzorz) [reply XRTK.SDK/Packages/com.xrtk.sdk/Features/Input/Handlers/ManipulationHandler.cs]: Let me rephrase: could we move this out?

Comment (StephenHodgson) [reply XRTK.SDK/Packages/com.xrtk.sdk/Features/Input/Handlers/ManipulationHandler.cs]: Yeah I'm not liking how this property is setup atm. But the purpose of all the logic is to ensure that the object is properly setup, this ensures we always will get back out a valid value, if the dependencies aren't setup. It's always understood that a Property can be doing additional work. That's part of what makes encapsulation so powerful, and why many people frown upon directly accessing public fields unless they're readonly or constant.

File: XRTK.SDK/Packages/com.xrtk.sdk/Features/Input/Handlers/ManipulationHandler.cs [hunk 2]
```diff
@@ -896,8 +905,8 @@ public virtual void BeginHold(MixedRealityPointerEventData eventData, Vector3? g
             }
 
             PrimaryPointer.OverrideGrabPoint = grabOffset;
-
-            transform.SetCollidersActive(false);
+            transform.GetColliderBounds(ref cachedColliders);
```
Comment (StephenHodgson) [inline XRTK.SDK/Packages/com.xrtk.sdk/Features/Input/Handlers/ManipulationHandler.cs]:
```suggestion
```
